# Карта социальных проблем сельских поселений Белгородского района

**Дашборд**: https://datalens.yandex.cloud/vvtsxpo6z66uh-karta-socialnyh-problem-selskih-territoriy?tab=ZL&state=a0ca17f3146

Основные цели проекта:
1. **Выявление мнений и потребностей жителей**
   - Исследовать, как жители оценивают текущую ситуацию в поселении.
   - Определить достижения, которые вызывают гордость, и проблемы, требующие решения.

2. **Анализ препятствий для объединения сообщества**
   - Изучить факторы, мешающие жителям объединяться, чтобы разработать меры по улучшению взаимодействия между ними.

3. **Подготовка рекомендаций для местной власти и НКО**
   - На основе анализа данных сформулировать рекомендации по повышению качества жизни в поселении.

## Предобработка данных

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)

In [3]:
#try:
answers = pd.read_excel('/Users/ksenia/Desktop/survey_data/2024 Данные анкеты и Сельские поселения.xlsx')
#except:
#answers = pd.read_excel('')

In [4]:
township = pd.read_excel('/Users/ksenia/Desktop/survey_data/2024 Данные анкеты и Сельские поселения.xlsx', sheet_name = 'Сельские поселения'\
                        , header=1)

In [5]:
township.head()

,№,Поселение,Сокр,"Площадь, тыс. кв.км","Доля, %","Население, чел.","Доля, %.1",Плотность,ТОС,"Доля, %.2",Жителей на территории ТОС,Члены ТОС,Активисты ТОС,Стобец,Строка
0,11,Беловское с/п,Бело,60.719,0.041764,6058,0.031769,99.771077,12.0,0.060302,2173.0,1689.0,120.0,4,4
1,10,Беломестненское с/п,Белм,38.800,0.026688,6092,0.031947,157.010309,6.0,0.030151,1948.0,150.0,120.0,3,2
2,13,Бессоновское с/п,Бес,164.232,0.112962,4777,0.025051,29.086901,5.0,0.025126,1410.0,152.0,108.0,1,6
3,15,Веселолопанское с/п,ВЛоп,40.888,0.028124,3593,0.018842,87.874193,11.0,0.055276,3270.0,274.0,33.0,2,7
4,18,Головинское с/п,Гол,93.210,0.064112,2362,0.012387,25.340629,4.0,0.020101,875.0,137.0,28.0,2,8


In [6]:
answers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3116 entries, 0 to 3115
Data columns (total 66 columns):
 #   Column                                                                                                                                                                                                     Non-Null Count  Dtype 
---  ------                                                                                                                                                                                                     --------------  ----- 
 0   ID                                                                                                                                                                                                         3116 non-null   int64 
 1   Время создания                                                                                                                                                                                     

In [7]:
answers.head()

,ID,Время создания,1.1. Сельское поселение,"1.2 В какой мере вас интересует то, что происходит в Вашем поселении?",1.3 Как Вы в целом оцениваете качество жизни в Вашем поселении?,"1.4 Как Вы считаете, учитывается ли Ваше мнение при принятии решений о развитие Вашего поселения?",1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом? / сельская администрация,1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом? / муниципальная администрация,1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом? / областная администрация,"1.6 От кого, по вашему мнению, в наибольшей степени зависит благополучие поселения?",1.6 Ваша оценка ситуации в поселении,1.7 Чем можно гордиться жителям Вашего поселения?,"2.1 Как Вы считаете, какие проблемы сейчас наиболее актуальны для жителей Вашего поселения?",2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / дошкольного образования,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / школьного образования,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / здравоохранения,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / культурного досуга,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / возможности для занятий спортом,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / коммунальных услуг,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / пассажирских перевозок,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / содержания и состояния дорог,2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / бытового обслуживания,"2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / социальное обслуживание, социальная поддержка и защита граждан","2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка семьи, материнства, отцовства и детства","2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / охрана здоровья граждан, пропаганда здорового образа жизни","2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка проектов в области науки, образования, просвещения",2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка проектов в области культуры и искусства,2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / сохранение исторической памяти,2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / выявление и поддержка молодых талантов,2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / охрана окружающей среды и защита животных,"2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / защита прав и свобод человека и гражданина, в том числе защита прав заключенных",2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / укрепление межнационального и межрелигиозного согласия,2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Дети дошкольного возраста,2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Дети младшего школьного возраста,2.4 Оцените по шкале от 1 до 5 

In [8]:
# пропуски в столбцах
answers.isna().sum()

ID                                                                                                                            0
Время создания                                                                                                                0
1.1. Сельское поселение                                                                                                       0
1.2 В какой мере вас интересует то, что происходит в Вашем поселении?                                                         0
1.3 Как Вы в целом оцениваете качество жизни в Вашем поселении?                                                               0
                                                                                                                           ... 
4.4 Какие формы решения проблем поселения Вам близки? / Подготовка социальных проектов                                      342
4.4 Какие формы решения проблем поселения Вам близки? / Организация общих культурных, спортивных, детски

Выводы:
- В столбцах 3.6, 4.4 присутствуют пропуски, что нормально, потому что эти вопросы подразумевают множественный выбор из нескольких ответов.
- Также присутствует большое количество пропусков в открытых вопросах, поэтому чтобы не забывать про них заменим их на "нет ответа".

## Поиск дубликатов

In [9]:
# есть два похожих столбца, проверим одинаковы ли в них значения
are_equal = (answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры'] \
             == answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры.1']).all()
display('Все значения одинаковы:', are_equal)

'Все значения одинаковы:'

False

In [10]:
mismatches = answers[answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры']\
                     != answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры.1']]
display('Строки с различиями:')
display(mismatches[['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры', '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры.1']])

'Строки с различиями:'

,2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры,2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры.1
4,4,3
9,3,5
13,4,3
38,4,5
41,3,4
...,...,...
3085,2,3
3091,3,5
3107,3,4
3109,3,2


In [11]:
# количество строк с различиями
mismatches.shape[0]

354

In [12]:
answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры'].mean()

4.271822849807445

In [13]:
answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры'] \
= np.ceil(answers[['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры',\
                   '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры.1']]\
          .mean(axis=1)).astype(int)

In [14]:
answers['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры'].mean()

4.316752246469833

In [15]:
answers = answers.drop(columns=['2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры.1'])

## Обработка пропусков

In [16]:
columns_to_fill = ['1.6 Ваша оценка ситуации в поселении',
                   '1.7 Чем можно гордиться жителям Вашего поселения?',
                   '2.1 Как Вы считаете, какие проблемы сейчас наиболее актуальны для жителей Вашего поселения?',
                   '4.3. Что, по Вашему мнению, препятствует объединению местного (соседского) сообщества?'
                  ]

In [17]:
# так как пропусков много не будем их удалять, а заменим на "нет ответа"
answers[columns_to_fill] = answers[columns_to_fill].fillna('нет ответа')

In [18]:
answers[columns_to_fill].isna().sum()

1.6 Ваша оценка ситуации в поселении                                                           0
1.7 Чем можно гордиться жителям Вашего поселения?                                              0
2.1 Как Вы считаете, какие проблемы сейчас наиболее актуальны для жителей Вашего поселения?    0
4.3. Что, по Вашему мнению, препятствует объединению местного (соседского) сообщества?         0
dtype: int64

## Статистический анализ результатов исследования

In [19]:
# функция для подсчета количества ответов и их процента
def answers_counts(column_name):
    results = (answers[column_name]
               .value_counts()
               .reset_index()
               .rename(columns={column_name: 'Ответ'})
               .rename(columns={'count': 'Количество'})  
               .assign(Процент=lambda x: (x['Количество'] / x['Количество'].sum() * 100)))
    return(results)

### Общая информация

In [20]:
answers_counts('3.1 Ваш пол')

,Ответ,Количество,Процент
0,Женский,2798,89.794608
1,Мужской,318,10.205392


- Данные показывают, что подавляющее большинство участников опроса – женщины.

In [21]:
answers_counts('3.2 Возраст')

,Ответ,Количество,Процент
0,36-45,1355,43.485237
1,26-35,778,24.967908
2,46-55,478,15.340180
3,56-65,264,8.472401
4,66-75,108,3.465982
5,19-25,78,2.503209
6,76 и старше,28,0.898588
7,18 и младше,27,0.866496


- Преобладание возрастной группы 36-45 лет: Эта группа составляет наибольшее количество респондентов — почти половина (43.49%), что может указывать на активное участие людей в этом возрасте в проводимом исследовании.
- Вторая по численности группа — 26-35 лет, что также свидетельствует о значительном представительстве молодых профессионалов.
- Снижение представительства в старших возрастных группах: После 55 лет наблюдается резкое снижение количества респондентов, особенно в группах 66-75 и 76 и старше.
- Минимальное представительство молодежи и подростков: групп 18 лет и младше и 19-25 лет выделяется существенно меньше.

In [22]:
answers_counts('3.3 Образование')

,Ответ,Количество,Процент
0,высшее,1807,57.991014
1,среднее профессиональное,681,21.854942
2,более одного высшего,306,9.820282
3,среднее,226,7.252888
4,незаконченное высшее,96,3.080873


- Более половины (57.99%) респондентов имеют высшее образование, что говорит о довольно высоком образовательном уровне участников исследования.
- Значительное количество со средним профессиональным образованием: Практически каждый пятый респондент имеет среднее профессиональное образование (21.85%), что также является важным показателем.
- Меньшее представительство для остальных категорий: Остальные категории (более одного высшего, среднее, незаконченное высшее) представляют собой значительно меньшие группы, что может указывать на то, что опрос был более ориентирован на образованных людей.

In [23]:
answers_counts('3.4 Сфера Вашей деятельности')

,Ответ,Количество,Процент
0,сотрудник бюджетной организации,1384,44.415918
1,сотрудник коммерческой структуры,736,23.620026
2,домохозяйка/домохозяин,448,14.377407
3,пенсионер,273,8.761232
4,предприниматель,145,4.653402
5,безработный,58,1.861361
6,студент/учащийся,45,1.444159
7,сотрудник НКО,27,0.866496


- Преобладание сотрудников бюджетной сферы: Более 44% респондентов работают в бюджетных организациях, что может указывать на наличие стабильной занятости и определенную социальную ответственность в ответах этой группы.
- Солидное представительство сотрудников коммерческих структур: Подсчет говорит о том, что почти четверть респондентов работает в коммерческих компаниях, что может отражать самозанятость и предпринимательский подход.
- Достаточно высокая доля домохозяйств: Более 14% респондентов являются домохозяевами, что может свидетельствовать о интересе к вопросам, связанным с семьей и бытом.
- Меньшее представительство пенсионеров и студентов: У пенсионеров (8.76%) и студентов/учащихся (1.44%) достаточно низкие показатели, что может говорить о меньшей вовлеченности этих групп в опрос.

In [24]:
answers_counts('3.7 Как Вы можете оценить финансовое благополучие своей семьи?')

,Ответ,Количество,Процент
0,"3. Денег хватает ровно на повседневную жизнь (еда, транспорт, коммуналка)",1747,56.065469
1,"2. В целом, денег хватает на все основные нужды и остаётся, чтобы отдохнуть",831,26.668806
2,4. Денег не хватает даже на повседневные нужды,205,6.578947
3,6. Затрудняюсь ответить,179,5.744544
4,5. Денег катастрофически не хватает,83,2.663671
5,"1. Денег достаточно для благополучной жизни, можем позволить себе новый транспорт или недвижимость",71,2.278562


- На основе представленных данных видно, что наибольшее количество людей (56.07%) считает, что «денег хватает ровно на повседневную жизнь». Это может указывать на то, что многие семьи находятся на «грани» и живут от зарплаты до зарплаты.
- Следующая по численности группа (26.67%) отвечает, что «в целом, денег хватает на все основные нужды и остаётся, чтобы отдохнуть». Это говорит о более комфортном финансовом положении, но всё равно подчеркивает, что не все чувствуют себя хорошо обеспеченными.
- Лишь 6.58% респондентов ответили, что «денег не хватает даже на повседневные нужды», что является положительным показателем. Это указывает на то, что большинство семей могут покрыть хотя бы основы.
- 5.74% затруднились ответить. Это может указывать на неопределенность или отсутствие ясности в их финансовой ситуации.
- 2.66% респондентов чувствуют, что «денег катастрофически не хватает». Это важно учитывать, так как даже небольшая доля таких ответов может означать, что часть населения испытывает значительное финансовое давление.
- Наконец, всего 2.28% уверены, что им хватает денег «для благополучной жизни и покупки новой недвижимости или транспорта». Это сигнализирует о том, что лишь небольшой процент семей находится в финансово стабильном положении, позволяющем им делать крупные покупки.

### 1.2 В какой мере вас интересует то, что происходит в Вашем поселении?

In [25]:
# уникальные ответы
answers['1.2 В какой мере вас интересует то, что происходит в Вашем поселении?'].unique()

array(['очень волнует', 'постольку поскольку', 'не интересует совсем'],
      dtype=object)

Создадим категории ответов с числовыми значениями, чтобы сравнить уровень интереса жителей по различных категориям (Сельское поселение, Пол, Возраст, Образование, Сфера деятельности, Тип семьи, Фин. состояние).

In [26]:
# функция для преобразования текстовых ответов в числовые значения
def interest_index(response):
    if response == 'очень волнует':
        return 1
    elif response == 'постольку поскольку':
        return 0
    elif response == 'не интересует совсем':
        return -1
    else:
        return None  # на случай, если ответ не соответствует ожидаемым

In [27]:
answers['1.2 индекс В какой мере вас интересует то, что происходит в Вашем поселении?'] = \
answers['1.2 В какой мере вас интересует то, что происходит в Вашем поселении?'].apply(interest_index)

In [28]:
answers_counts('1.2 В какой мере вас интересует то, что происходит в Вашем поселении?')

,Ответ,Количество,Процент
0,очень волнует,2553,81.931964
1,постольку поскольку,537,17.233633
2,не интересует совсем,26,0.834403


In [29]:
# сумма ответов
round(answers['1.2 индекс В какой мере вас интересует то, что происходит в Вашем поселении?'].sum()/answers.shape[0], 3)

0.811

#### Выводы
**Высокий уровень интереса:**

81.93% респондентов отметили, что их очень волнует то, что происходит в поселении. Это указывает на высокий уровень вовлеченности и заинтересованности жителей в местных вопросах. Уровень интереса может говорить о желании участвовать в жизни своего сообщества, а также о том, что жители готовы к диалогу и сотрудничеству с властями.

**Умеренный интерес:**

17.23% опрошенных сказали, что их интерес к происходящему «постольку поскольку». Эта категория респондентов может стремиться к большему вовлечению, но пока не проявляет активного желания участвовать в жизни поселения. Возможно, им нужно больше информации или стимула для участия в местных событиях или инициативах.

**Низкий уровень интереса:**

0.83% респондентов не интересуются происходящим в поселении. Эта доля достаточно мала, что свидетельствует о низком уровне апатии среди жителей. Однако все же стоит обратить внимание на эту группу, чтобы понять, почему они не заинтересованы, и работать над тем, чтобы повысить их вовлеченность.

### 1.3 Как Вы в целом оцениваете качество жизни в Вашем поселении?

In [30]:
# уникальные ответы
answers['1.3 Как Вы в целом оцениваете качество жизни в Вашем поселении?'].unique()

array(['высокое', 'среднее', 'выше среднего', 'низкое', 'ниже среднего'],
      dtype=object)

In [31]:
# функция для преобразования текстовых ответов в числовые значения
def quality_life(response):
    if response == 'низкое':
        return -1
    elif response == 'ниже среднего':
        return -0.5
    elif response == 'среднее':
        return 0
    elif response == 'выше среднего':
        return 0.5
    elif response == 'высокое':
        return 1
    else:
        return None  # на случай, если ответ не соответствует ожидаемым

In [32]:
answers['1.3 индекс Как Вы в целом оцениваете качество жизни в Вашем поселении?'] = \
answers['1.3 Как Вы в целом оцениваете качество жизни в Вашем поселении?'].apply(quality_life)

In [33]:
answers_counts('1.3 Как Вы в целом оцениваете качество жизни в Вашем поселении?')

,Ответ,Количество,Процент
0,среднее,1759,56.450578
1,выше среднего,699,22.432606
2,ниже среднего,299,9.595635
3,низкое,206,6.611040
4,высокое,153,4.910141


In [34]:
# сумма ответов
round(answers['1.3 индекс Как Вы в целом оцениваете качество жизни в Вашем поселении?'].sum()/answers.shape[0], 3)

0.047

#### Выводы
Основная часть опрашиваемых считает качество жизни средним, что может говорить о том, что есть как положительные, так и отрицательные аспекты, влияющие на общее восприятие.

**Среднее качество жизни:**

56.45% респондентов оценили качество жизни как "среднее". Это говорит о том, что большинство жителей считает свою жизнь в поселении на приемлемом уровне, однако не высоком. Возможно, это указывает на наличие проблем, которые требуют внимания.

**Выше среднего:**

22.43% респондентов оценили качество жизни как "выше среднего". Это положительный сигнал, указывающий на то, что значительная часть населения видит достоинства жизни в поселении и может свидетельствовать о наличии успешных инициатив или комфортной среды.

**Ниже среднего:**

9.60% респондентов оценили качество жизни как "ниже среднего". Это указывает на определенные недостатки и проблемы, которые могут требовать решения и внимания со стороны местной администрации.

**Низкое и высокое качество жизни:**

6.61% оценили качество жизни как "низкое", а 4.91% - как "высокое". Эти показатели показывают, что явных крайностей в восприятии качества жизни нет, однако, и те, и другие доли достаточно малочисленны.

### 1.4 Как Вы считаете, учитывается ли Ваше мнение при принятии решений о развитие Вашего поселения?

In [35]:
# уникальные ответы
answers['1.4 Как Вы считаете, учитывается ли Ваше мнение при принятии решений о развитие Вашего поселения?'].unique()

array(['да', 'нет', 'затрудняюсь ответить'], dtype=object)

In [36]:
# функция для преобразования текстовых ответов в числовые значения
def importance_opinion(response):
    if response == 'да':
        return 1
    elif response == 'затрудняюсь ответить':
        return 0
    elif response == 'нет':
        return -1
    else:
        return None  # на случай, если ответ не соответствует ожидаемым

In [37]:
answers['1.4 индекс Как Вы считаете, учитывается ли Ваше мнение при принятии решений о развитие Вашего поселения?'] = \
answers['1.4 Как Вы считаете, учитывается ли Ваше мнение при принятии решений о развитие Вашего поселения?'].apply(importance_opinion)

In [38]:
column_name = '1.4 индекс Как Вы считаете, учитывается ли Ваше мнение при принятии решений о развитие Вашего поселения?'

results = (answers[column_name]
           .value_counts()
           .reset_index()
           .rename(columns={column_name: 'Ответ'})
           .rename(columns={'count': 'Количество'})  
           .assign(Процент=lambda x: (x['Количество'] / x['Количество'].sum() * 100)))
print(results)

   Ответ  Количество    Процент
0      0        1362  43.709884
1     -1        1163  37.323492
2      1         591  18.966624


#### Выводы
- 43.7% жителей воздержались от ответа, это довольно высокая доля респондентов, которые либо не имеют мнения по этому вопросу, либо не хотят или не могут выразить его. Это может указывать на: низкий уровень информированности о процессах принятия решений в местном управлении, непонимание значения вопроса или его актуальности для них, cкептическое или равнодушное отношение к тому, как их мнение может повлиять на развитие поселения;
- 37.3% ответили, что их мнение не учитывается, эта группа выражает недовольство и чувство исключенности из процессов принятия решений. Такой вывод говорит о: cуществующих проблемах в коммуникации между жителями и местными властями, ощущении безразличия со стороны власти к мнению жителей, возможной необходимости в улучшении механизма обратной связи или участия населения в процессе принятия решений.
- 19% считают, что их мнение учитывается. Этот процент относительно мал и может свидетельствовать о том, что лишь небольшая часть общественности уверена в том, что их мнения влияют на развитие поселения.

На основании этих данных можно сделать вывод о наличии значительного недовольства среди жителей по поводу отсутствия учёта их мнения.
Высокий процент воздержавшихся от ответа может быть сигналом о необходимости повышения информированности и вовлеченности населения.
Локальным властям может потребоваться активнее работать над улучшением коммуникации, вовлечения общественности в процесс принятия решений и реализации механизмов обратной связи, чтобы повысить уровень доверия и участия граждан в управлении поселенческими вопросами.

### 1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом?

In [39]:
mean_scores = answers.groupby('1.1. Сельское поселение')[[
    '1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом? / сельская администрация',
    '1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом? / муниципальная администрация',
    '1.5 Оцените по шкале от 1 до 5 уровень доверия и взаимодействия между властью и обществом? / областная администрация'
]].mean().round(2).reset_index()
mean_scores.columns = ['Поселение', 'Сельская администрация', 'Муниципальная администрация)'\
                       , 'Областная администрация']
mean_scores

,Поселение,Сельская администрация,Муниципальная администрация),Областная администрация
0,Беловское с/п,3.31,3.36,3.45
1,Беломестненское с/п,3.20,3.16,3.23
2,Бессоновское с/п,3.44,3.49,3.55
3,Веселолопанское с/п,3.58,3.26,3.23
4,Головинское с/п,3.68,3.42,3.73
5,Дубовское с/п,3.43,3.28,3.30
6,Ериковское с/п,3.58,3.38,3.50
7,Журавлевское с/п,4.00,3.60,3.00
8,Комсомольское с/п,4.29,3.90,3.62
9,Краснооктябрьское с/п,3.44,3.21,3.29


#### Выводы
- В большинстве случаев средние рейтинги областной администрации выше, чем у муниципальной и сельской.

### 1.6 От кого, по вашему мнению, в наибольшей степени зависит благополучие поселения?

In [40]:
answers['1.6 От кого, по вашему мнению, в наибольшей степени зависит благополучие поселения?'].unique()

array(['От региональных властей', 'От каждого жителя',
       'От поселковой администрации',
       'От сплоченности местного сообщества', 'От федеральных властей',
       'другое', 'От местных муниципальных властей',
       'От местного бизнеса'], dtype=object)

In [41]:
answers_counts('1.6 От кого, по вашему мнению, в наибольшей степени зависит благополучие поселения?')

,Ответ,Количество,Процент
0,От поселковой администрации,803,25.770218
1,От каждого жителя,571,18.324775
2,От региональных властей,542,17.394095
3,От местных муниципальных властей,456,14.634146
4,От сплоченности местного сообщества,412,13.222080
5,От федеральных властей,225,7.220796
6,другое,85,2.727856
7,От местного бизнеса,22,0.706033


#### Выводы
- Наиболее распространенное мнение о том, что поселковая администрация оказывает наибольшее влияние на благополучие, стоит использовать для обоснования местных инициатив и программ по улучшению качества жизни.
- Необходимо разработать больше инициатив, направленных на усиление сплоченности сообщества, например, путем организации общественных мероприятий или программ, которые способствуют взаимодействию между жителями.

### 1.6 Ваша оценка ситуации в поселении

In [42]:
# определяем функцию для категоризации ответов на вопрос "1.6 Ваша оценка ситуации в поселении"
def categorize_response_rating(response):
    keywords_positive = ['хоро', 'устраив', 'позитивн', 'красив', 'нравит', 'стабил', 'достойн' , 'положит', 'отличн','благоприят'\
                         , 'высок', 'спокойн', 'благополуч', 'прекрасн', 'лучшая'\
                         ,'замечат', 'достойн', 'чисто', 'комфорт', 'выше средн', 'шикарно', 'мирная', 'благоустр'\
                         , 'слажено', '4', '5', '6', '7', '8', '9','чет']
    keywords_positive_num  = ['10','100']
    keywords_negative = ['сделай', 'вод', 'остановк', 'плох', 'проблем', 'не хватает', 'негативно', 'напряж', 'мусор', 'беспок', 'удруч'\
                         , 'бесхоз', 'некомп', 'неудовл', 'отвратит', 'мало', 'не хватает', 'критич', 'нужд', 'низк', 'сложн', 'плач'\
                         , 'брош', 'ужас', 'нет', 'оставляет', 'недоволен', 'тревож', 'ниже', 'невынос', 'отсутств', 'сво', 'печал'\
                         , 'не очень', 'войн', 'имитац', 'так себе', 'вред', 'администр', 'невозможн', 'нестаб', 'страшно', 'укрыт', 'сложн'\
                         , 'дорог', 'не просто', 'гряз', 'власт', 'бардак', 'опасн', 'угроз', 'чс', 'умир', 'неудовлетворительная', 'беспок'\
                         , 'бедств', 'тяж', 'не безопасно', 'безразлич', 'недостаточно', 'воняет', 'неочень', 'не хотят', 'отрицат', 'риск'\
                         ,'неблагополуч', 'выжив', 'нехватка', 'убеж', 'урадническая', 'что хотят', 'трудн', 'не развив', 'воен'\
                         ,'ничего не делается', 'все хуже', 'некто не хочет', 'собак', 'не выполн', 'как и везде', 'не хватает', 'необходимо'\
                         ,'разрушен','2','1','0','два','нулев', 'массив 62.25', 'очень сложная']
    keywords_no_answ = ['нет ответа', 'воздержус', 'затрудн', 'никаково', 'ни какого', 'незнаю', 'Не знаю что писать']
    
    # приводим ответ к нижнему регистру для удобства
    response_lower = response.lower()
    
    # проверяем ключевые слова
    if any(keyword in response_lower for keyword in keywords_no_answ):
        return 'Нет ответа'
    elif any(keyword in response_lower for keyword in keywords_positive_num):
        return 'Позитивная'
    elif any(keyword in response_lower for keyword in keywords_negative):
        return 'Негативная'
    elif any(keyword in response_lower for keyword in keywords_positive):
        return 'Позитивная'
    else:
        return 'Нейтральная'

In [43]:
answers['1.6 категория Ваша оценка ситуации в поселении'] = answers['1.6 Ваша оценка ситуации в поселении'].apply(categorize_response_rating)

In [44]:
# функция для преобразования текстовых ответов в числовые значения
def rating_number(response):
    if response == 'Позитивная':
        return 1
    elif response == 'Нейтральная':
        return 0
    elif response == 'Негативная':
        return -1
    else:
        return None  # на случай, если ответ не соответствует ожидаемым

In [45]:
answers['1.6 индекс Ваша оценка ситуации в поселении'] = answers['1.6 категория Ваша оценка ситуации в поселении'].apply(rating_number)

In [46]:
answers.groupby('1.6 индекс Ваша оценка ситуации в поселении')['1.6 индекс Ваша оценка ситуации в поселении'].count()

1.6 индекс Ваша оценка ситуации в поселении
-1.0    776
 0.0    598
 1.0    558
Name: 1.6 индекс Ваша оценка ситуации в поселении, dtype: int64

In [47]:
answers_counts('1.6 категория Ваша оценка ситуации в поселении')

,Ответ,Количество,Процент
0,Нет ответа,1184,37.997433
1,Негативная,776,24.903723
2,Нейтральная,598,19.191271
3,Позитивная,558,17.907574


#### Выводы
**Нет ответа (37.99%)**
- Наибольшее количество респондентов не дало ответа на этот вопрос. Это может означать, что жители не уверены в своей оценке ситуации, либо не хотят ее высказывать.
**Негативная оценка (24.90%)**
- Четверть респондентов оценивают ситуацию в поселении негативно. Это может указывать на существующие проблемы, такие как недостаток инфраструктуры, ограниченные возможности для досуга, или плохие условия жизни.
- Это сигнал для местных властей о том, что необходимо провести оценку текущих проблем и разработать стратегии их решения.
**Нейтральная оценка (19.19%)**
- Приблизительно 19% респондентов имеют нейтральное мнение о ситуации. Это может означать, что они не видят существенных изменений как в положительном, так и в отрицательном направлении.
**Позитивная оценка (17.91%)**
- Самая низкая доля (менее 18%) респондентов оценивает ситуацию в поселении положительно. Это подчеркивает, что жителей в общем не удовлетворяет текущая обстановка.

Негативные оценки ситуации указывают на существующие проблемы и необходимость оперативных действий со стороны властей. Увеличение позитивного восприятия может быть достигнуто через инициативы, направленные на улучшение качества жизни, создание новых возможностей и активное взаимодействие с жителями.

### 1.7 Чем можно гордиться жителям Вашего поселения

In [48]:
# определяем функцию для категоризации ответов на вопрос "1.7 Чем можно гордиться жителям Вашего поселения"
def categorize_response_pride(response):
    categories_dict = {
        '1.7 Благоустройство и качество городской среды': 0,
        '1.7 Медицина': 0,
        '1.7 Спорт': 0,
        '1.7 Транспорт': 0,
        '1.7 Образование': 0,
        '1.7 Природа и окружающая среда': 0,
        '1.7 Социальная жизнь и сообщество': 0,
        '1.7 История, традиции и культура': 0,
        '1.7 Администрация': 0,
        '1.7 Негативные или скептические отзывы': 0,
        '1.7 Нет ответа': 0,
        '1.7 Другие': 0
    }
    
    keywords_infrastructure = ['благоус', 'чист', 'сквер', 'парк', 'отдых', 'инфраст', 'интерьр', 'магазин', 'площад', 'озелен'\
                               , 'парк', 'район', 'ухож', 'учрежд', 'безопас', 'территор', 'уют', 'порядок', 'cоциал', 'красотой села'\
                               , 'опрятност', 'поряд', 'обустройств', 'газ', 'улиц', 'архитектур', 'соц', 'инфаструк'\
                               , 'внешн', 'благополуч', 'досуг', 'двор', 'пятероч', 'развит', 'фабрик', 'расширен', 'ресторан', 'жизн']
    keywords_medicine = ['медиц', 'поликлин', 'реабилитац', 'доктор', 'медпункт', 'амбулат']
    keywords_sport = ['стадион', 'ксц', 'спорт', 'арен', 'бассейн', 'фок']
    keywords_transport = ['транспорт', 'трасс', 'автобус', 'cветофор', 'дорог']
    keywords_education = ['школ', 'сад','гимназ', 'образов', 'сош', 'институт']
    keywords_nature  = ['природ', 'зел', 'пруд', 'эколог', 'закат', 'родник', 'местност', 'воздух', 'речк', 'рыб', 'руч', 'погод']
    keywords_people = ['люд', 'жител', 'сообщест', 'сплоч', 'объедин', 'друж', 'волонт', 'отзывч', 'дружелюб', 'сосед', 'помощь'\
                       , 'земляк', 'защищ', 'сил', 'стойкост', 'терпен', 'сплачен', 'народ', 'общен', 'мужествен', 'отважн', 'ребят'\
                       , 'молодеж', 'уроженц', 'выходц', 'отношени', 'пенсионер', 'взаимоуважен', 'помог', 'инициатив', 'гражд', 'населен'\
                       , 'единств', 'общ', 'талант', 'безразлич', 'зашищает', 'жив', 'cмелост', 'идея', 'успех', 'трудолюб', 'достижен'\
                       , 'смелост', 'сектор', 'взаимопониман']
    keywords_history = ['истори', 'культ', 'традиц', 'родин', 'прошл', 'геро', 'cтойк', 'горин', 'былым', 'памят', 'отечеств', 'музе'\
                        , 'храм', 'достопримечат', 'церкв', 'музе', 'библиотек', 'елисеев', 'ссср', 'предк', 'памятник', 'слав'\
                        , 'церков', 'достопремечат', 'цкр']
    keywords_administration = ['администр', 'глав', 'управл', 'губернатор', 'управлени', 'председат', 'власт']
    keywords_no_answ = ['нет ответа', 'воздержус', 'затрудн', 'незнаю', 'не знаю', 'ничего']
    keywords_nothing = ['ничем', 'ни чем', 'не чем', 'нечем', 'мусор', 'бардак', 'помойк', 'тяжело']
    
    #приводим ответ к нижнему регистру для удобства
    response_lower = response.lower()

    # проверяем категории
    found_any_category = False

    if any(keyword in response_lower for keyword in keywords_infrastructure):
        categories_dict['1.7 Благоустройство и качество городской среды'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_medicine):
        categories_dict['1.7 Медицина'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_sport):
        categories_dict['1.7 Спорт'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_transport):
        categories_dict['1.7 Транспорт'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_education):
        categories_dict['1.7 Образование'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_nature):
        categories_dict['1.7 Природа и окружающая среда'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_people):
        categories_dict['1.7 Социальная жизнь и сообщество'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_history):
        categories_dict['1.7 История, традиции и культура'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_administration):
        categories_dict['1.7 Администрация'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_nothing):
        categories_dict['1.7 Негативные или скептические отзывы'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_no_answ):
        categories_dict['1.7 Нет ответа'] = 1
        found_any_category = True
    # установим категории "Другие", если ни одна категория не была найдена
    if not found_any_category:
        categories_dict['1.7 Другие'] = 1
    return categories_dict

In [49]:
# применяем функцию и добавляем результаты в DataFrame
categories_df = answers['1.7 Чем можно гордиться жителям Вашего поселения?'].apply(categorize_response_pride).apply(pd.Series)
answers = pd.concat([answers, categories_df], axis=1)

### 2.1 Как Вы считаете, какие проблемы сейчас наиболее актуальны для жителей Вашего поселения?

In [50]:
# определяем функцию для категоризации ответов на вопрос 
# "Как Вы считаете, какие проблемы сейчас наиболее актуальны для жителей Вашего поселения?"
def categorize_response_locality_problem(response):
    categories_dict = {
        '2.1 Транспорт': 0,
        '2.1 Медицина': 0,
        '2.1 Образование': 0,
        '2.1 Коммунальные услуги': 0,
        '2.1 Связь и интернет': 0,
        '2.1 Инфраструктура и благоустройство': 0,
        '2.1 Работа и благостояние': 0,
        '2.1 Безопасность': 0,
        '2.1 Культура, отдых и досуг': 0,
        '2.1 Административные проблемы': 0,
        '2.1 Нет ответа': 0,
        '2.1 Другие': 0
     }
    
    keywords_transport = ['дорог', 'транспорт', 'тротуар', 'автобус', 'пешеход', 'тратуар', 'асфальт', 'дорож', 'парковк' ]
    keywords_medicine = ['больниц','поликлиник', 'медпункт', 'поликлиника', 'аптек', 'здравоохранение', 'медицин','здоровье'\
                         , 'медобслуживание', 'медперсонал', 'врач']
    keywords_education = ['школ','образов', 'детский сад', 'дет сад', 'обучение', 'детски', 'сад']
    keywords_utilities = ['вода', 'водоснабж','водоотвод', 'воды','освещени', 'свет', 'утилизац', 'отход', 'вывоз', 'мумор', 'мусор'\
                          , 'уборка', 'тко']
    keywords_communication = ['связ', 'интернет']
    keywords_infrastructure = [ 'инфраструктур', 'кладбищ', 'магазин', 'благоустройств', 'супермаркет', 'банкомат', 'инфростуктура']
    keywords_ecology = ['загрязн', 'зелен', 'эколог', 'гряз', 'очистн', 'деревь', 'запах', 'парк', 'лес', 'вонь', 'запах'\
                        , 'река', 'речк', 'воздух']
    keywords_job = ['безработиц', 'работа', 'социальн', 'рабоч', 'работ', 'ваканси','доход', 'цена', 'цены', 'инвестиц', 'эконом'\
                    , 'оплата труда', 'зарплата']
    keywords_safety = ['преступност', 'освещение', 'охран', 'безопас', 'угроза', 'война', 'воен', 'сво', 'укрыти', 'обстрел', 'мир'\
                       , 'границ', 'защит', 'обстановк', 'приграничн', 'беспилотник', 'грани', 'дрон', 'бпла', 'страх', 'всу', 'спокойств', 'сирен'\
                       , 'враг', 'опасность', 'военная', 'боевые', 'беспилотны', 'бездомные', 'собак']
    keywords_culture = ['культур', 'отдых', 'кружк', 'досуг', 'мероприят', 'спорт', 'клуб', 'стадион', 'молодеж', 'бассейн', 'дк'\
                       , 'библиотек', 'центр', 'храм', 'творчество', 'площад']
    keywords_administration = ['взаимодействие', 'граждан', 'администраци', 'власт']
    keywords_no_answ = ['нет ответа', 'воздержус', 'затрудн', 'незнаю', 'не знаю', 'ничего', 'хз', 'не могу ответить ']
    
    # приводим ответ к нижнему регистру для удобства
    response_lower = response.lower()

    # проверяем категории
    found_any_category = False
    
    # проверяем ключевые слова
    if any(keyword in response_lower for keyword in keywords_transport):
        categories_dict['2.1 Транспорт'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_medicine):
        categories_dict['2.1 Медицина'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_education):
        categories_dict['2.1 Образование'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_utilities):
        categories_dict['2.1 Коммунальные услуги'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_communication):
        categories_dict['2.1 Связь и интернет'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_infrastructure):
        categories_dict['2.1 Инфраструктура и благоустройство'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_job):
        categories_dict['2.1 Работа и благостояние'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_safety):
        categories_dict['2.1 Безопасность'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in  keywords_culture):
        categories_dict['2.1 Культура, отдых и досуг'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in  keywords_administration):
        categories_dict['2.1 Административные проблемы'] = 1
        found_any_category = True
    if any(keyword in response_lower for keyword in keywords_no_answ):
        categories_dict['2.1 Нет ответа'] = 1
        found_any_category = True
    if not found_any_category:
        categories_dict['2.1 Другие'] = 1
    return categories_dict

In [51]:
# применяем функцию и добавляем результаты в DataFrame
categories_locality_problem_df = answers['2.1 Как Вы считаете, какие проблемы сейчас наиболее актуальны для жителей Вашего поселения?']\
.apply(categorize_response_locality_problem).apply(pd.Series)
answers = pd.concat([answers, categories_locality_problem_df], axis=1)

### 4.1 В каких мероприятиях Вашего поселения Вы принимали участие в последнее время (2-3 месяца)?

In [52]:
answers_counts('4.1 В каких мероприятиях Вашего поселения Вы принимали участие в последнее время (2-3 месяца)?')

,Ответ,Количество,Процент
0,ни в каких,1187,38.093710
1,субботники,576,18.485237
2,культурные мероприятия,505,16.206675
3,собрания жителей,496,15.917843
4,другое,237,7.605905
5,спортивные праздники,115,3.690629


#### Выводы
**Ни в каких (38.09%):**

- Наибольшее количество респондентов (38.09%) сообщает, что не участвовало ни в каких мероприятиях за указанный период. Это высокий процент, который может указывать на низкий уровень вовлеченности жителей в жизнь поселения. Возможно, существует недостаток информации о мероприятиях или интереса к ним среди жителей. Это требование к местным властям и организаторам мероприятий – увеличивать информированность о возможностях участия.

**Субботники (18.49%):**

- Участие в субботниках на уровне 18.49% сигнализирует о том, что жители готовы работать на благо своего поселения, но этот показатель также невысок. Это может быть признаком того, что жители готовы участвовать в активностях, направленных на улучшение окружающей среды, но необходимо больше вовлечения.

**Культурные мероприятия (16.21%):**

- Показатель участия в культурных мероприятиях составляет 16.21%. Это говорит о наличии интереса к культурной жизни, но также указывает на то, что доступны еще не все возможные форматы, чтобы привлечь больше людей.

**Собрания жителей (15.92%):**

- Участие в собраниях жителей (15.92%) говорит о том, что есть интерес к обсуждению местных вопросов. Однако этот показатель тоже не гарантирует, что жители будут активно вовлечены в принятие решений.

**Другое (7.61%):**

- Эта категория может включать различные мероприятия, не попадающие в вышеперечисленные категории. Это открывает возможность для дополнительного сбора информации о том, какие именно мероприятия интересуют жителей.

**Спортивные праздники (3.69%):**

- Наименьшее число участников проявило интерес к спортивным праздникам. Это может указывать на недостаток спортивных мероприятий или их организационных форм в поселении, а возможно, и на меньшее внимание к физической активности среди жителей.

**Общие выводы:**
- Низкий уровень вовлеченности: Высокий процент людей, которые не участвуют в мероприятиях, указывает на возможные проблемы с вовлечением жителей в социальную жизнь поселения.
- Инициативы по повышению участия: Существующие мероприятия, такие как субботники и культурные события, могут быть улучшены для более активного привлечения населения.
- Информирование: Улучшение информирования о мероприятиях и их значимости для жизни поселения может увеличить уровень участия.

**Рекомендации**

- Создание коммуникационной стратегии: Открыты и ясны каналы для информирования жителей о предстоящих мероприятиях. Это могут быть социальные сети, местные собрания, объявления на сайте.
- Организация дополнительных мероприятий: Проведение более разнообразных мероприятий, учитывающих интересы жителей, может способствовать повышению их вовлеченности.
- Работа с сообщества: Привлечение местных лидеров мнений или активистов для организации мероприятий может способствовать увеличению участия и интереса к жизни поселения.

### 4.2 По Вашему мнению, способны ли жители поселения объединиться для совместного решения местных задач?

In [53]:
answers['4.2 По Вашему мнению, способны ли жители поселения объединиться для совместного решения местных задач?'].unique()

array(['Возможно', 'Уже объединены', 'Нет'], dtype=object)

In [54]:
# функция для преобразования текстовых ответов в числовые значения
def association(response):
    if response == 'Уже объединены':
        return 1
    elif response == 'Возможно':
        return 0
    elif response == 'Нет':
        return -1
    else:
        return None  # на случай, если ответ не соответствует ожидаемым

In [55]:
answers['4.2 индекс По Вашему мнению, способны ли жители поселения объединиться для совместного решения местных задач?'] = \
answers['4.2 По Вашему мнению, способны ли жители поселения объединиться для совместного решения местных задач?'].apply(association)

In [56]:
answers_counts('4.2 По Вашему мнению, способны ли жители поселения объединиться для совместного решения местных задач?')

,Ответ,Количество,Процент
0,Возможно,2550,81.835687
1,Уже объединены,327,10.494223
2,Нет,239,7.670090


#### Выводы
**Возможно (81.84%):**

- Большая часть респондентов (81.84%) считает, что жители способны объединиться для решения местных задач. Это свидетельствует о высоком уровне надежды на возможность совместного действия и о готовности к сотрудничеству.
- Такой оптимизм может быть использован для инициирования местных инициатив и программ, которые будут способствовать развитию сообщества, так как есть доверие к тому, что жители готовы работать вместе.

**Уже объединены (10.49%):**

- Небольшая доля (10.49%) людей утверждает, что жители уже объединены для решения местных задач. Это может указывать на существующие инициативные группы или проекты по улучшению жизни в поселении.
- Важно поддерживать и развивать эти группы, выделять их достижения, чтобы показать жителям, что объединение возможно и приводит к реальным результатам.

**Нет (7.67%):**

- Лишь небольшая часть респондентов (7.67%) считает, что жители не способны объединиться. Это мнение стоит учитывать, но оно представляет собой меньшинство.

Высокий процент респондентов, поддерживающих идею о возможности объединения, создает крепкую основу для развития совместных инициатив и проектов. Существующие инициативные группы и проекты следует развивать и поддерживать, чтобы продемонстрировать успешные примеры объединения и сотрудничества среди жителей.

### 4.3. Что, по Вашему мнению, препятствует объединению местного (соседского) сообщества?

In [57]:
# определяем функцию для категоризации ответов на вопрос 
# "Что, по Вашему мнению, препятствует объединению местного (соседского) сообщества?"
def categorize_response_community_problem(response):
    keywords_situation = ['обстановк', 'ситуаци', 'сво','враждеб', 'стран', 'война', 'опасн', 'обстрел', 'сирен', 'военные действия'\
                         , 'нынешние реалии']
    keywords_reluctance = ['желани', 'заинтерес', 'лень', 'безразлич', 'мотивац', 'хата с краю', 'пофигизм', 'безъиниативн', 'безинициатив'\
                          ,'врем', 'некогда', 'занят', 'работ', 'нехотение', 'активное участие', 'пассивность', 'интерес', 'активность'\
                          , 'инициатив']
    keywords_strangers = ['не знают', 'не знаком', 'малообщит', 'дружб', 'отчужден', 'сплочён', 'доверие', 'общени', 'разобщен', 'разобщ'
                          , 'разобщён', 'разрознен', 'сплаченност', 'взгляд', 'мнени', 'разноглас', 'озлоблен', 'индивидуал', 'менталитет'\
                          , 'воспитан', 'умение слушать', 'приезж', 'эгоизм', 'друг к другу', 'коммуникац', 'личност', 'атомизация'\
                          , 'мировоззрение', 'только о себе', 'совместн', 'обида', 'гордые', 'объединен', 'коллектив', 'общительность'\
                          , 'конфликт', 'сам за себя', 'алкоголизм', 'свои проблемы', 'сам по себе', 'своей жизнью', 'малознакомы'\
                          , 'минталитет', 'алкаш', 'пьяниц', 'классовое разделение', 'общаться', 'жестокие', 'доброта', 'зависть', 'статус']
    keywords_organization = ['администрац', 'алминистрация', 'губернатор', 'управлени', 'организац', 'организовать ', 'лидер', 'человек'\
                             , 'курирован', 'руководит', 'глава поселения', 'главы поселения','организа', 'активист', 'чиновник'\
                             , 'вел за собой', 'предводител', 'информ', 'неосведомл', 'оповещен', 'уличком', 'никто не реагирует', 'инициатив'\
                             , 'организовать', 'финансиров', 'ответственность', 'нас не слышат', 'официальное лицо','власть','власти'\
                             , 'никто не пробовал', 'чат', 'публикац', 'нет предложений', 'места притяжения', 'организованность	']
    keywords_no_answ = ['нет ответа', 'воздержус', 'затрудн', 'незнаю', 'не знаю', 'ничего', 'хз', 'не могу ответить ']
    
    # приводим ответ к нижнему регистру для удобства
    response_lower = response.lower()
    if any(keyword in response_lower for keyword in keywords_organization):
        return 'Лидерство и управление'
    elif any(keyword in response_lower for keyword in keywords_reluctance):
        return 'Мотивация и время'
    elif any(keyword in response_lower for keyword in keywords_strangers):
        return 'Социальные связи и коммуникация'
    elif any(keyword in response_lower for keyword in keywords_situation):
        return 'Обстановка и безопасность'
    elif any(keyword in response_lower for keyword in keywords_no_answ):
        return 'Нет ответа'
    else:
        return 'Другие'

In [58]:
answers['4.3. категория Что, по Вашему мнению, препятствует объединению местного (соседского) сообщества?'] =\
answers['4.3. Что, по Вашему мнению, препятствует объединению местного (соседского) сообщества?'].apply(categorize_response_community_problem)

### 4.5 Сколько времени в неделю Вы смогли бы выделить на общественную работу на безвозмездных условиях?

In [59]:
answers['4.5 И последний вопрос: сколько времени в неделю Вы смогли бы выделить на общественную работу на безвозмездных условиях?'].unique()

array(['Несколько часов в неделю',
       'Нисколько, очень много работы и семейных забот',
       '1-2 дня (не подряд, а в общей сложности)',
       'Нисколько, бесплатно принципиально не работаю',
       'Большую часть времени готов заниматься общественными делами'],
      dtype=object)

In [60]:
# функция для преобразования текстовых ответов в числовые значения
def time(response):
    if response == 'Нисколько, бесплатно принципиально не работаю':
        return -1
    elif response == 'Нисколько, очень много работы и семейных забот':
        return -0.5
    elif response == 'Несколько часов в неделю':
        return 0.5
    elif response == '1-2 дня (не подряд, а в общей сложности)':
        return 1
    elif response == 'Большую часть времени готов заниматься общественными делами':
        return 1.5
    else:
        return None  # на случай, если ответ не соответствует ожидаемым

In [61]:
answers['4.5 индекс И последний вопрос: сколько времени в неделю Вы смогли бы выделить на общественную работу на безвозмездных условиях?'] = \
answers['4.5 И последний вопрос: сколько времени в неделю Вы смогли бы выделить на общественную работу на безвозмездных условиях?'].apply(time)

In [62]:
answers_counts('4.5 И последний вопрос: сколько времени в неделю Вы смогли бы выделить на общественную работу на безвозмездных условиях?')

,Ответ,Количество,Процент
0,Несколько часов в неделю,1422,45.635430
1,"Нисколько, очень много работы и семейных забот",1155,37.066752
2,"1-2 дня (не подряд, а в общей сложности)",354,11.360719
3,"Нисколько, бесплатно принципиально не работаю",136,4.364570
4,Большую часть времени готов заниматься общественными делами,49,1.572529


#### Выводы
**Несколько часов в неделю (45.64%):**

- Почти половина респондентов (45.64%) готова выделить несколько часов в неделю на общественную работу. Это указывает на наличие значительного числа людей, которые заинтересованы в участии в местных инициативах. Это может стать основой для формирования групп для реализации общественных проектов, где можно использовать это время на благо сообщества.

**Нисколько, очень много работы и семейных забот (37.07%):**

- Более трети респондентов (37.07%) заявляют, что не могут уделять время общественной работе из-за нагрузки на работе и семейных обязанностей. Это подчеркивает важность создания таких условий, при которых жители смогут участвовать в жизни поселения, не испытывая дополнительных стрессов.

**1-2 дня (не подряд, а в общей сложности) (11.36%):**

- Несколько жителей (11.36%) готовы выделить 1-2 дня в общей сложности на общественную работу.

**Нисколько, бесплатно принципиально не работаю (4.36%):**

- Небольшой процент (4.36%) людей не считает возможным участие в общественной работе на безвозмездной основе. Это может быть связано с убеждениями или личными приоритетами.

**Большую часть времени готов заниматься общественными делами (1.57%):**

- Меньшинство респондентов (1.57%) готово посвятить большую часть своего времени общественной работе. Это говорит о высокой степени преданности некоторой части жителей, и их опыт и энергия могут быть полезны организаторам.

**Общие выводы:**
- Значительная доля респондентов готова участвовать в общественной работе, что является положительным знаком для сообщества.
- Учитывая, что многие люди испытывают нехватку времени из-за работы и семейных обязанностей, важно создать гибкие форматы добровольного участия.
- Необходимо поддерживать и развивать инициативные группы, привлекающие тех, кто хочет участвовать больше, что может помочь решить проблемы, стоящие перед обществом.

**Рекомендации**

- Гибкие форматы участия: Разработка программ и мероприятий, которые требуют минимальных временных затрат и могут быть проведены в удобное для жителей время.
- Информирование о возможностях: Создание информационных ресурсов, где жители могли бы видеть, какие общественные инициативы существуют и какие из них требуют помощи и участия.
- Поддержка и мотивация: Реализация программ, которые поощряют участие, например, объявление о достижениях и благодарности участникам за их вклад, может повысить уровень вовлеченности.

### Преобразование в длинный формат и сохранение

In [63]:
answers_list = ['ID','3.6 Как можно охарактеризовать Вашу семью? / обычная'\
        , '3.6 Как можно охарактеризовать Вашу семью? / многодетная'\
        , '3.6 Как можно охарактеризовать Вашу семью? / многодетная'\
        , '3.6 Как можно охарактеризовать Вашу семью? / малообеспеченная'\
        , '3.6 Как можно охарактеризовать Вашу семью? / молодая'\
        , '3.6 Как можно охарактеризовать Вашу семью? / семья с одаренными детьми'\
        , '3.6 Как можно охарактеризовать Вашу семью? / семья с детьми с инвалидностью и ОВЗ'\
        , '3.6 Как можно охарактеризовать Вашу семью? / семья и профессиональных трудовых династий'\
        , '3.6 Как можно охарактеризовать Вашу семью? / неполная'\
        , '3.6 Как можно охарактеризовать Вашу семью? / неженат / холост'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / дошкольного образования'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / школьного образования'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / здравоохранения'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / культурного досуга'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / возможности для занятий спортом'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / коммунальных услуг'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / пассажирских перевозок'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / содержания и состояния дорог'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / бытового обслуживания'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / социальное обслуживание, социальная поддержка и защита граждан'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка семьи, материнства, отцовства и детства'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / охрана здоровья граждан, пропаганда здорового образа жизни'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка проектов в области науки, образования, просвещения'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка проектов в области культуры и искусства'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / сохранение исторической памяти'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / выявление и поддержка молодых талантов'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / охрана окружающей среды и защита животных'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / защита прав и свобод человека и гражданина, в том числе защита прав заключенных'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / укрепление межнационального и межрелигиозного согласия'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Дети дошкольного возраста'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Дети младшего школьного возраста'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Подростки'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Молодежь'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Люди с инвалидностью'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Многодетные семьи'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Неполные семьи'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в общественных организациях'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в субботниках, благоустройстве территории'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Подготовка социальных проектов'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Организация общих культурных, спортивных, детских мероприятий'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в сходах граждан'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в охране общественного порядка (ДНД)'\
        , '1.7 Благоустройство и качество городской среды', '1.7 Медицина', '1.7 Спорт', '1.7 Транспорт'\
        , '1.7 Образование', '1.7 Природа и окружающая среда', '1.7 Социальная жизнь и сообщество'\
        , '1.7 История, традиции и культура', '1.7 Администрация', '1.7 Негативные или скептические отзывы', '1.7 Нет ответа'\
        , '1.7 Другие', '2.1 Транспорт', '2.1 Медицина', '2.1 Образование', '2.1 Коммунальные услуги', '2.1 Связь и интернет'\
        , '2.1 Инфраструктура и благоустройство', '2.1 Работа и благостояние', '2.1 Безопасность', '2.1 Культура, отдых и досуг'\
        , '2.1 Административные проблемы', '2.1 Нет ответа', '2.1 Другие']

In [64]:
answers_category = answers[answers_list]

In [65]:
answers_long = pd.melt(answers_category, id_vars = ['ID'], var_name = 'Вопрос', value_name = 'Ответ')

In [66]:
columns_to_delete = ['2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / дошкольного образования'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / школьного образования'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / здравоохранения'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / культурного досуга'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / возможности для занятий спортом'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / коммунальных услуг'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / пассажирских перевозок'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / содержания и состояния дорог'\
        , '2.2 Оцените по шкале от 1 до 5 качество предоставляемых услуг в вашем поселении в сфере / бытового обслуживания'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / социальное обслуживание, социальная поддержка и защита граждан'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка семьи, материнства, отцовства и детства'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / охрана здоровья граждан, пропаганда здорового образа жизни'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка проектов в области науки, образования, просвещения'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / поддержка проектов в области культуры и искусства'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / сохранение исторической памяти'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / выявление и поддержка молодых талантов'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / охрана окружающей среды и защита животных'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / защита прав и свобод человека и гражданина, в том числе защита прав заключенных'\
        , '2.3 Оцените по шкале от 1 до 5 актуальность в Вашем поселении изменений (социальных проектов) по следующим направлениям / укрепление межнационального и межрелигиозного согласия'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Дети дошкольного возраста'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Дети младшего школьного возраста'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Подростки'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Молодежь'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Пенсионеры'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Люди с инвалидностью'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Многодетные семьи'\
        , '2.4 Оцените по шкале от 1 до 5 какой категории населения особенно нужна дополнительная поддержка в Вашем поселении / Неполные семьи'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в общественных организациях'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в субботниках, благоустройстве территории'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Подготовка социальных проектов'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Организация общих культурных, спортивных, детских мероприятий'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в сходах граждан'\
        , '4.4 Какие формы решения проблем поселения Вам близки? / Участие в охране общественного порядка (ДНД)']

In [67]:
answers.drop(columns = columns_to_delete, inplace=True)

In [68]:
answers_long.to_csv('answers_long.csv', index=False, encoding='utf-8-sig')
answers.to_csv('answers.csv', index=False, encoding='utf-8-sig')
township.to_csv('.csv', index=False, encoding='utf-8-sig')

### Общий вывод

На основе проведенного исследования можно сделать несколько ключевых выводов о состоянии жизни в поселении и вовлеченности жителей в общественные дела. Большинство респондентов проявляют интерес к событиям, происходящим в их округе, однако уровень участия в мероприятиях остается низким, что может свидетельствовать о недостаточной информации или недостаточно привлекательном формате этих мероприятий. Оценка качества жизни неоднозначна: часть жителей довольна существующими условиями, но есть и немало критиков, что указывает на необходимость улучшения инфраструктуры и услуг. 
Мнения жителей относительно учета их позиций при принятии решений показывают, что многие чувствуют себя игнорируемыми местными властями, что негативно сказывается на доверии и взаимодействии между гражданами и государственными органами. Многие респонденты верят, что благополучие поселения зависит прежде всего от власти, в то время как некоторые подчеркивают важность участия самих граждан.

Качество предоставляемых услуг также требует внимания, поскольку многие жители отмечают недостатки. Существующий интерес к социальным проектам и необходимая поддержка определенных категорий населения указывают на желание жителей улучшить социальную среду. Хотя уровень участия в мероприятиях остается низким, у значительной части опрошенных есть готовность выделять время на безвозмездную общественную работу, что создает потенциал для более активного вовлечения населения в решении местных задач.

В целом, результаты опроса показывают необходимость повышения прозрачности власти, организации регулярного диалога с жителями, улучшения качества услуг и создания более привлекательных форматов для участия граждан в жизни сообщества. Это поможет укрепить доверие, повысить уровень вовлеченности и создать более активную, сплоченную общественную среду.

### Рекомендации

**Увеличение информированности:**
- Разработать и внедрить онлайн-платформы (веб-сайты, мобильные приложения) для информирования жителей о мероприятиях, социальных проектах и возможностях доступа к услугам.
- Использовать каналы электронной почты и социальные сети для регулярных оповещений о происходящих событиях, новшествах и изменениях в жизни поселения.
  
**Проведение встреч:**
- Проводить регулярные встречи между местными властями и жителями для обсуждения актуальных вопросов, планов и потребностей населения. Это поможет жителям чувствовать свою значимость и влияние на принимаемые решения.

**Улучшение качества жизни:**
- Уделять внимание благоустройству общественных мест – парков, площадей, спортивных объектов, чтобы сделать их более привлекательными для жителей.
- Инвестиции в инфраструктуру.

**Предоставление грантов:**
- Оказывать финансовую поддержку жителям, желающим реализовать инновационные идеи или проекты в интересах сообщества.

**Развитие волонтерского движения:**
- Вводить программы поощрения для волонтеров, например, предоставляя сертификаты, благодарности или небольшие материальные вознаграждения.
- Проводить тренинги для желающих стать волонтерами, чтобы повысить их навыки и уверенность в своих силах для участия в жизни сообщества.

**Поддержка уязвимых групп населения:**
- Разрабатывать специальные инициативы для помощи уязвимым категориям граждан, таким как семьи с низким доходом, пожилые люди или молодежь.

**Гибкие форматы участия:**
- Предлагать жителям разные способы участия (онлайн-встречи, очные мероприятия, рабочие группы) для того, чтобы каждый мог выбрать подходящий для себя вариант.
- Проводить мероприятия в удобное для жителей время и учитывать их занятость в расписании